cli/cmd/org/rename.go

```diff
@@ -26,65 +24,71 @@ func RenameCmd(ch *cmdutil.Helper) *cobra.Command {
 				return err
 			}
 
-			fmt.Println("Warn: Renaming an org would invalidate dashboard URLs")
-
-			if !cmd.Flags().Changed("org") && ch.Interactive {
-				orgNames, err := OrgNames(ctx, ch)
-				if err != nil {
-					return err
+			resp, err := client.GetOrganization(ctx, &adminv1.GetOrganizationRequest{Name: name})
+			if err != nil {
+				if st, ok := status.FromError(err); ok && st.Code() == codes.NotFound {
+					ch.PrintfError("org %q doesn't exist, run 'rill org list' to see available orgs", name)
+					return nil
 				}
+				return err
+			}
+			org := resp.Organization
 
-				name, err = cmdutil.SelectPrompt("Select org to rename", orgNames, "")
-				if err != nil {
-					return err
-				}
+			// Build update request
+			req := &adminv1.UpdateOrganizationRequest{Name: org.Name}
+
+			if req.NewName != nil && req.DisplayName == nil && org.DisplayName != "" {
+				ch.PrintfWarn("\nWarn: Changing org name will invalidate dashboard URLs.\n")
 			}
 
-			if ch.Interactive {
-				err = cmdutil.SetFlagsByInputPrompts(*cmd, "new-name")
-				if err != nil {
-					return err
-				}
+			var flagSet bool
+			if cmd.Flags().Changed("new-name") {
+				flagSet = true
+				req.NewName = &newName
 			}
 
-			if newName == "" {
-				return fmt.Errorf("please provide valid org new-name, provided: %q", newName)
+			if cmd.Flags().Changed("display-name") {
+				flagSet = true
+				req.DisplayName = &displayName
 			}
 
-			if !force {
-				msg := fmt.Sprintf("Do you want to rename org \"%s\" to \"%s\"?", color.YellowString(name), color.YellowString(newName)) // nolint:gocritic // Because it uses colors
-				ok, err := cmdutil.ConfirmPrompt(msg, "", false)
-				if err != nil {
-					return err
-				}
-				if !ok {
-					return nil
-				}
+			if !flagSet {
+				ch.PrintfError("no changes requested please specify --new-name or --display-name\n")
+				return nil
 			}
 
-			updatedOrg, err := client.UpdateOrganization(ctx, &adminv1.UpdateOrganizationRequest{
-				Name:    name,
-				NewName: &newName,
-			})
+			// Update org
+			updatedOrg, err := client.UpdateOrganization(ctx, req)
 			if err != nil {
 				return err
 			}
 
-			err = dotrill.SetDefaultOrg(newName)
-			if err != nil {
-				return err
+			// Print results
+			ch.PrintfSuccess("Updated organization\n")
+			if req.NewName != nil {
+				ch.Printf("Updated name: %s to %s\n", name, updatedOrg.Organization.Name)
+			}
```

Comment:
For example something like this for letting the user know that they may also want to update the display name:

```suggestion
			if req.NewName != nil {
				ch.Printf("Updated name %q to %q\n", name, updatedOrg.Organization.Name)
				if org.DisplayName != "" && req.DisplayName == nil {
					ch.Warnf("You updated the org's unique name, but not its display name; use --display-name to update the display name\n")
				}
			}
```